src/ledger/LedgerManager.h

```diff
@@ -129,6 +133,8 @@ class LedgerManager
     // Ledger txn here is needed for the sake of lazy load; it won't be
     // used most of the time.
     virtual SorobanNetworkConfig const& getSorobanNetworkConfig() = 0;
+    virtual SorobanNetworkConfig const& getSorobanNetworkConfigForApply() = 0;
```

Comment:
`resolveBackgroundEvictionScan` needs to call `getSorobanNetworkConfigForApply`. If the network config settings changes mid ledger, we need to invalidate the previous scan and start over based on the new settings. Currently this won't happen since we're using the RO snapshot and not the apply time snapshot. `startBackgroundEvictionScan` should also probably use `getSorobanNetworkConfigForApply` just for consistency (although `getSorobanNetworkConfig` should have the same value at that point anyway). Note that while the eviction scan does most it's work in the background, when it's joined at the resolution step, it should be considered part of TX apply.